commcare_connect/opportunity/tests/test_views.py

```diff
@@ -38,3 +49,84 @@ def test_add_budget_existing_users(
     assert opportunity.total_budget == 205
     assert opportunity.claimed_budget == 15
     assert OpportunityClaimLimit.objects.get(pk=ocl.pk).max_visits == 15
+
+
+class TestUserVisitReviewView:
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        client: Client,
+        program_manager_org: Organization,
+        program_manager_org_user_admin: User,
+        organization: Organization,
+        org_user_admin: User,
+    ):
+        self.client = client
+        self.pm_org = program_manager_org
+        self.pm_user = program_manager_org_user_admin
+        self.nm_org = organization
+        self.nm_user = org_user_admin
+        self.program = ProgramFactory(organization=self.pm_org)
+        self.opportunity = ManagedOpportunityFactory(program=self.program, organization=self.nm_org)
+        access = OpportunityAccessFactory(opportunity=self.opportunity, accepted=True)
+        self.visits = UserVisitFactory.create_batch(
+            10,
+            opportunity=self.opportunity,
+            status=VisitValidationStatus.approved,
+            review_created_on=now(),
+            review_status=VisitReviewStatus.pending,
+            opportunity_access=access,
+        )
+
+    def test_user_visit_review_program_manager_pending(self):
+        self.url = reverse("opportunity:user_visit_review", args=(self.pm_org.slug, self.opportunity.id))
+        self.client.force_login(self.pm_user)
+        response = self.client.get(self.url)
+        assert response.status_code == 200
+        table = response.context["table"]
+        assert len(table.rows) == 10
+        assert "pk" in table.columns.names()
+
+        visits = UserVisit.objects.filter(id__in=[visit.id for visit in self.visits])
+        for visit in visits:
+            assert visit.review_status == VisitReviewStatus.pending
+
+    def test_user_visit_review_program_manager_agree(self):
```

Comment:
You might be able to use `@pytest.mark.parametrize` to combine the agree and reject test cases into a single test. And similarly program_manager_pending and network_manager get request tests.